packages/router/src/__tests__/setContextReuse.test.tsx

```diff
@@ -87,7 +87,7 @@ test("Doesn't destroy <Set> when navigating inside, but does when navigating bet
   act(() => navigate(routes.ctx3()))
   await waitFor(() => screen.getByText('3-initialSetValue'))
   act(() => navigate(routes.ctx4()))
-  await waitFor(() => screen.getByText('4-initialSetValue'))
+  await waitFor(() => screen.getByText('4-updatedSetValue')) // Because its wrapping the same context provider, it should retain the value
```

Comment:
This test confuses me and I think its wrong. Given that Route 4 is nested in a completely brand new SetContextProvider  it should have a fresh context to use. It should not reuse the same context as the routes above it. If a user wanted to use the same context then they would just nest the route inside the other context. So by doing this kinda merge go against react patterns

Reply:
The problem we have is that its a catch 22. We cannot **both** have it reset the context, but **also** never re-mount shared wraps.

Reply:
@Tobbe - may be worth trying in your version actually - you're doing something with the props that might make it more possible.

The gist of what we need to do:
1. Don't set the key to setId (like [this](https://github.com/redwoodjs/redwood/pull/9299/files#diff-aaaf969c8c446a8f428b54b2d5482767b6d92c7ab4771e68b14145cdbe6a9dddL183)). This is the bit that prevents remounting of shared wrap components between sets.
2. We'd have to generate a unique id for each wrapper

e.g. 
```
//... in util's recurse function
if (wrapFromCurrentSet) {
          wrapperComponentsArray = Array.isArray(wrapFromCurrentSet)
            ? wrapFromCurrentSet
            : [wrapFromCurrentSet]

          wrapperComponentsArray.forEach((wrapper, i) => {
            wrapper.wrapperId = `set-${setId}-wrap-${i}`
          })
```
and then use this wrappedId as a key. 

The problem in our PR here, is that in the [reduce](https://github.com/redwoodjs/redwood/pull/9299/files#diff-aaaf969c8c446a8f428b54b2d5482767b6d92c7ab4771e68b14145cdbe6a9dddR257) it will over-write the key, where as in yours it doesn't (I think)